lib/amqp/session.rb

```diff
@@ -930,9 +930,15 @@ def receive_frame(frame)
         begin
           receive_frameset(@frames[frame.channel])
         ensure
-          # for channel.close, frame.channel will be nil. MK.
-          clear_frames_on(frame.channel) if @frames[frame.channel]
+          begin
+            clear_frames_on(frame.channel) if @frames[frame.channel] # for channel.close, frame.channel will be nil. MK.
+          rescue Exception => ex
```

Comment:
`ensure_safe` does actually rescue `Exception`, it's a pretty intense save. I can lower this down to a `StandardError` though if you'd prefer.